eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -354,6 +354,8 @@ case class UpdateAddHtlc(channelId: ByteVector32,
                          tlvStream: TlvStream[UpdateAddHtlcTlv]) extends HtlcMessage with UpdateMessage with HasChannelId {
   val blinding_opt: Option[PublicKey] = tlvStream.get[UpdateAddHtlcTlv.BlindingPoint].map(_.publicKey)
 
+  val endorsement: Int = tlvStream.get[UpdateAddHtlcTlv.Endorsement].map(_.level).getOrElse(0)
```

Comment:
I'm not sure an endorsement of `0` is the same thing as not providing an endorsement at all...shouldn't we make that an `Option[Int]` to reflect that and handle those differently? Or should it be handled exactly the same way?

Reply:
Interesting, I find that a bit surprising. It seems to me that an incoming value of `0` means "I'm really afraid this HTLC will fail" whereas no incoming value means "I have no idea" and would rather be comparable to a value of 3.5? Are you really sure treating them the same way doesn't bias the reputation algorithm?

Reply:
The input values should be looked at relative to each other not in absolute terms. We could use any value here, as long as it's constant.

Reply:
Ok, I don't fully get that yet, but it's dependent on the actual reputation algorithm that will be used in the next PR, so it can be ignored for now.